tpm2/Makefile

```diff
@@ -386,65 +402,113 @@ kmyth-test: $(TEST_OBJECTS)
 
 $(TEST_UTIL_OBJ_DIR)/%.o: $(TEST_UTIL_SRC_DIR)/%.c \
-                          $(TEST_UTIL_INC_DIR)/%.h \
-                          | $(TEST_UTIL_OBJ_DIR)
+                          $(TEST_UTIL_INC_DIR)/%.h | \
+                          $(TEST_UTIL_OBJ_DIR)
 	$(CC) $(KMYTH_CFLAGS) \
 	      $(KMYTH_INCLUDE_FLAGS) \
 	      $(TEST_INCLUDE_FLAGS) \
 	      $< \
 	      -o $@
 
 $(TEST_TPM_OBJ_DIR)/%.o: $(TEST_TPM_SRC_DIR)/%.c \
-                         $(TEST_TPM_INC_DIR)/%.h \
-                         | $(TEST_TPM_OBJ_DIR)
+                         $(TEST_TPM_INC_DIR)/%.h | \
+                         $(TEST_TPM_OBJ_DIR)
 	$(CC) $(KMYTH_CFLAGS) \
 	      $(KMYTH_INCLUDE_FLAGS) \
 	      $(TEST_INCLUDE_FLAGS) \
 	      $< -o \
 	      $@
 
 $(TEST_CIPHER_OBJ_DIR)/%.o: $(TEST_CIPHER_SRC_DIR)/%.c \
-                            $(TEST_CIPHER_INC_DIR)/%.h \
-                            | $(TEST_CIPHER_OBJ_DIR)
+                            $(TEST_CIPHER_INC_DIR)/%.h | \
+                            $(TEST_CIPHER_OBJ_DIR)
 	$(CC) $(KMYTH_CFLAGS) \
 	      $(KMYTH_INCLUDE_FLAGS) \
 	      $(TEST_INCLUDE_FLAGS) \
 	      $< \
 	      -o $@
 
+$(TEST_OBJ_DIR):
+	mkdir -p $(TEST_OBJ_DIR)
+
+$(TEST_MAIN_OBJ_DIR):
+	mkdir -p $(TEST_MAIN_OBJ_DIR)
+
+$(TEST_UTIL_OBJ_DIR):
+	mkdir -p $(TEST_UTIL_OBJ_DIR)
+
+$(TEST_TPM_OBJ_DIR):
+	mkdir -p $(TEST_TPM_OBJ_DIR)
+
+$(TEST_CIPHER_OBJ_DIR):
+	mkdir -p $(TEST_CIPHER_OBJ_DIR)
+
 .PHONY: install
 install:
+ifeq ($(wildcard $(LOGGER_LIB_LOCAL_DEST)), $(LOGGER_LIB_LOCAL_DEST))
 	install -d $(DESTDIR)$(PREFIX)/lib
-	install -m 755 $(LIB_DIR)/$(KMYTH_LIB_SONAME) $(DESTDIR)$(PREFIX)/lib/
-	install -d $(DESTDIR)$(PREFIX)/include/kmyth/
+	install -m 755 $(LOGGER_LIB_LOCAL_DEST) $(DESTDIR)$(PREFIX)/lib/
+	install -d $(DESTDIR)$(PREFIX)/include/kmyth
 	install -m 644 $(LOGGER_INC_DIR)/kmyth_log.h \
 	               $(DESTDIR)$(PREFIX)/include/kmyth/
+	ldconfig
+endif
+ifeq ($(wildcard $(TPM_UTIL_LIB_LOCAL_DEST)), $(TPM_UTIL_LIB_LOCAL_DEST))
+	install -d $(DESTDIR)$(PREFIX)/lib
+	install -m 755 $(TPM_UTIL_LIB_LOCAL_DEST) $(DESTDIR)$(PREFIX)/lib/
+	install -d $(DESTDIR)$(PREFIX)/include/kmyth
 	install -m 644 $(INC_DIR)/kmyth.h $(DESTDIR)$(PREFIX)/include/kmyth/
+	ldconfig
+endif
+ifeq ($(wildcard $(BIN_DIR)/kmyth-seal), $(BIN_DIR)/kmyth-seal)
 	install -d $(DESTDIR)$(PREFIX)/bin
 	install -m 755 $(BIN_DIR)/kmyth-seal $(DESTDIR)$(PREFIX)/bin/
+endif
+ifeq ($(wildcard $(BIN_DIR)/kmyth-unseal), $(BIN_DIR)/kmyth-unseal)
+	install -d $(DESTDIR)$(PREFIX)/bin
 	install -m 755 $(BIN_DIR)/kmyth-unseal $(DESTDIR)$(PREFIX)/bin/
-	ldconfig
+endif
 
 .PHONY: uninstall
 uninstall:
-	rm -f $(DESTDIR)$(PREFIX)/lib/$(KMYTH_LIB_SONAME)
-	rm -f $(DESTDIR)$(PREFIX)/include/kmyth/kmyth_log.h
-	rm -f $(DESTDIR)$(PREFIX)/include/kmyth/kmyth.h
+	rm -f $(DESTDIR)$(PREFIX)/lib/$(TPM_UTIL_LIB_SONAME)
+	rm -f $(DESTDIR)$(PREFIX)/lib/$(LOGGER_LIB_SONAME)
+	rm -rf $(DESTDIR)$(PREFIX)/include/kmyth
```

Comment:
If someone has installed the sgx portion of the tool, this will unintentionally remove that as well. This should remove specific headers.

Reply:
Good point ... I changed to delete the specific headers and then the directory, only if it is empty